tests/conftest.py

```diff
@@ -219,8 +218,8 @@ def config(
 
 
 @pytest.fixture()
-def config_dir(tmp_dir: str) -> Path:
-    return Path(tempfile.mkdtemp(prefix="poetry_config_", dir=tmp_dir))
+def config_dir(tmp_path: Path) -> Path:
+    return Path(tempfile.mkdtemp(prefix="poetry_config_", dir=tmp_path))
```

Comment:
```suggestion
    path = tmp_path / "config"
    path.mkdir()
    return path
```

I don't think we need a random name here since we create it in the `tmp_path` for each test.

Reply:
:heavy_check_mark: 
Should I review the usage of tempfile in tests and use tmp_path where possible as well?

Reply:
It's ok for me to put this of until later. You can see if it's worth in a follow up PR if you like.